### PR TITLE
Question 2 Solved

### PR DESCRIPTION
**Pull request by taruns52**
Please check Question 2 code and let me know if there is any issue

### CODE REVIEW

File: Question_2.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+dict = {}
+
+# empty dictionary to store the frequency count for each element (key), 
+# and its corresponding value is a list with all elements that have this key's value
+for name in names:
+    if len(name) in dict:
+        dict[len(name)] += 1
+    else:
+        dict[len(name)] = 1
```
Comment (himanshu-suthar-tagline) [inline Question_2.py]: You can do here like:

dict[len(name)] = dict.get(len(name), 0) + 1

Also, please rename the `dict` to something else, as `dict` is reserved keyword in python which is used for `dict` class. 

Make sure you don't define variables that are reserved keyword in python.

File: Question_2.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        values[1]: temp_names 
+    }
+
+    # format -> count of names of same length : { length of name : [names] }
+
+# printing data in required format
+print('\nThe three least frequent name lenghts are:\n',)
+for x in ans:
+    for key , value in ans[x].items():
+        print('{} names of length {}: {}'.format( key,x, value))
```
Comment (himanshu-suthar-tagline) [inline Question_2.py]: Can you able to optimize the code?

Comment (taruns52) [reply Question_2.py]: Yes, sir, I have optimized the code please review it and let me know if I can improve it further.

File: Question_2.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ans = []
+
+# loop through the keys and values from above dictionary
+# so that we can required answer in required format 
+for length in sorted(most_frequent_names.keys()):
+    temp_names = []
+
+    # get the number of times the current length occurs
+    for name in names:
+        if len(name)  == length:
```
Comment (himanshu-suthar-tagline) [inline Question_2.py]: Remove extra space after `len(name)`

File: Question_2.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+for values in rev_dict:
+    temp_names = []
+
+    # get the number of times the current length occurs
+    for name in names:
+        if len(name)  == values[0]:     #values[0] = length of name , values[1] = count of names with same length
+            temp_names.append(name)
+    
+    # creating the dictioanry named ans to store ans
+    ans.append([ values[1], values[0], temp_names ])
```
Comment (himanshu-suthar-tagline) [inline Question_2.py]: The code is repeated as same as above. Please create a function and use it for both places.

Comment (taruns52) [reply Question_2.py]: I have applied all the changes you have suggested and please review and let me know the areas of improvement.

**Pull request merged**